Foundation/NSTask.swift

```diff
@@ -194,11 +189,69 @@ public class NSTask : NSObject {
             argv.dealloc(args.count + 1)
         }
         
-        // Lock...
+        var taskSocketPair : [Int32] = [0, 0]
+        socketpair(AF_UNIX, SOCK_STREAM, 0, &taskSocketPair)
         
-        processLaunchLock.lock()
+        var context = CFSocketContext(version: 0, info: UnsafeMutablePointer<Void>(Unmanaged.passUnretained(self).toOpaque()),
+                                               retain: runLoopSourceRetain, release: runLoopSourceRelease, copyDescription: nil)
         
-        // ...and load
+        let socket = CFSocketCreateWithNative( nil, taskSocketPair[0], CFSocketCallBackType.DataCallBack.rawValue, {
```

Comment:
from my limited understanding is that free floating inline closures have a different storage than functions; specifically if I grok what is happening correctly it is the capture-ness of the closure that is the issue (I know we have had a few sneak around in Foundation so far that have caused heap/stack corruptions).

I would definitely like to be wrong on this so maybe one of the runtime/compiler team can comment with a more informed stance on this?